src/olympia/addons/cron.py

```diff
@@ -28,34 +27,36 @@
 task_log = olympia.core.logger.getLogger('z.task')
 
 
-def update_addon_average_daily_users():
+def update_addon_average_daily_users(chunk_size=250):
     """Update add-ons ADU totals."""
     if not waffle.switch_is_active('local-statistics-processing'):
         return False
 
-    # BigQuery does not have data for add-ons with type other than those in
-    # `ADDON_TYPES_WITH_STATS` so we use download counts instead.
-    # See: https://github.com/mozilla/addons-server/issues/14609
-    amo_counts = dict(
+    counts = dict(
+        # In order to reset the `average_daily_users` values of add-ons that
+        # don't exist in BigQuery, we prepare a set of `(guid, 0)` for most
+        # add-ons.
         Addon.objects
-        .exclude(type__in=amo.ADDON_TYPES_WITH_STATS)
+        .filter(type__in=amo.ADDON_TYPES_WITH_STATS)
         .exclude(guid__isnull=True)
         .exclude(guid__exact='')
-        .annotate(count=Coalesce(Sum('downloadcount__count'), 0))
+        .exclude(average_daily_users=0)
+        .annotate(count=Value(0, IntegerField()))
         .values_list('guid', 'count')
         # Just to make order predictable in tests, we order by id. This
         # matches the GROUP BY being generated so it should be safe.
         .order_by('id')
     )
-    counts = dict(get_addons_and_average_daily_users_from_bigquery())
-    counts.update(amo_counts)
+    # Update the `counts` with values from BigQuery.
+    counts.update(dict(get_addons_and_average_daily_users_from_bigquery()))
     counts = list(counts.items())
```

Comment:
If this new strategy works, I think we could apply it to the `hotness` cron task. We wouldn't have 2 different tasks anymore, we wouldn't have a giant `IN` clause and we the number of add-ons to process/tasks would be roughly the same.